sql/src/main/java/org/apache/druid/sql/calcite/planner/DruidPlanner.java

```diff
@@ -116,13 +127,26 @@ private PlannerResult planWithDruidConvention(
       final RelRoot root
   ) throws RelConversionException
   {
-    final DruidRel<?> druidRel = (DruidRel<?>) planner.transform(
-        Rules.DRUID_CONVENTION_RULES,
-        planner.getEmptyTraitSet()
-               .replace(DruidConvention.instance())
-               .plus(root.collation),
-        root.rel
-    );
+    final RelNode wrappedRootRel = possiblyWrapRootWithOuterLimitFromContext(root);
+
+    final DruidRel<?> druidRel;
+    if (wrappedRootRel == null) {
```

Comment:
could `possiblyWrapRootWithOuterLimitFromContext` just return `root.rel` instead of `null` to collapse these near identical planner transform blocks?